### PR TITLE
Adding auth_proxy role and install files

### PR DESCRIPTION
**Pull request by neelimamukiri**
The install_* and uninstall_* files allow triggering install and
cleanup actions for specific components. This allows the end user to
specify finer grained installation (only docker, only scheduler, only
contiv etc.)

### CODE REVIEW

File: roles/auth_proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+- hosts: netplugin-node
+  become: true
+      roles:
+        - { role: auth_proxy, auth_proxy_port: 10000, auth_proxy_datastore: etcd://netmaster:2379 }
+
+License
+-------
+
+BSD
```
Comment (jojimt) [inline roles/auth_proxy/README.md]: Why is this BSD?

File: uninstall_auth_proxy.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - include_vars: roles/{{ item }}/defaults/main.yml
+      with_items:
+        - "contiv_network"
+        - "contiv_storage"
+        - "swarm"
+        - "kubernetes"
+        - "ucp"
+        - "docker"
+        - "etcd"
+        - "auth_proxy"
```
Comment (jojimt) [inline uninstall_auth_proxy.yml]: Should we be referring to contiv_storage and ucp here?

File: roles/auth_proxy/README.md (hunk 1 above)
Comment (neelimamukiri) [reply roles/auth_proxy/README.md]: It started with the intent of making it an Ansible Galaxy role. I dont see any specific license for contiv_network. Should I remove the license field here?

File: uninstall_auth_proxy.yml (hunk 2 above)
Comment (neelimamukiri) [reply uninstall_auth_proxy.yml]: Ideally no. We have some cross references of defaults between docker, etcd and contiv. auth_proxy does not have any dependencies. I will remove the references to contiv_storage and ucp.

**Comment by jojimt** [comment]
LGTM

**Pull request merged**